src/group_theory/abelianization.lean

```diff
@@ -106,4 +111,58 @@ theorem hom_ext (φ ψ : abelianization G →* A)
   (h : φ.comp of = ψ.comp of) : φ = ψ :=
 monoid_hom.ext $ λ x, quotient_group.induction_on x $ monoid_hom.congr_fun h
 
+section map
+
+variables {H : Type v} [group H] (f : G →* H)
+
+/-- The map operation of the `abelianization` functor -/
+def map : abelianization G →* abelianization H := lift (of.comp f)
+
+variables {f}
+
+@[simp]
+lemma map_apply {x : abelianization G} : map f x = lift (of.comp f) x := rfl
+
+@[simp]
+lemma map_id : map (monoid_hom.id G) = monoid_hom.id (abelianization G) := hom_ext _ _ rfl
+
+lemma map_comp {I : Type w} [group I] {g : H →* I} :
+  (map g).comp (map f) = map (g.comp f) := hom_ext _ _ rfl
```

Comment:
It might be worth also having
```lean
  map g (map f x) = map (g.comp f) x := monoid_hom.congr_fun (map_comp _ _) x
```

Reply:
Linter complained
```
Error: abelianization.map_map_apply - Left-hand side simplifies from
  ⇑(abelianization.map g) (⇑(abelianization.map f) x)
to
  ⇑(⇑abelianization.lift (abelianization.of.comp g)) (⇑(⇑abelianization.lift (abelianization.of.comp f)) x)
using 
  [abelianization.map_apply]
Try to change the left-hand side to the simplified term!
```
should I 
1. Make `abelianization.map_map_apply` not @[simp]?
2. Remove the `@[simp]` from `map_apply` (so that `map f x` remains)
3. Reverse their order maybe?

Reply:
Option 2 sounds good to me